sync-core/src/main/java/com/cloudant/sync/datastore/encryption/CacheKeyProvider.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2013 Cloudant, Inc. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.cloudant.sync.datastore.encryption;
+
+import java.util.logging.Logger;
+
+/**
+ * This class implements the interface {@link KeyProvider} and it can be used to create an
+ * encrypted datastore.
+ * <p/>
```

Comment:
When I run `gradle docs` I get `error: self-closing element not allowed` - this may vary across Java versions so you might not get this.